custom_enumerable_methods.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    my_self = self
+    i = 0
+    while i < my_self.size
+      yield(my_self[i])
+      i += 1
+    end
+    my_self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = self
+    i = 0
+    element = 0
+    my_self.my_each do
+      yield(element, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_self = self
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?(*arg)
```

Comment:
If it is given a range it should return false if the block returns false or nil.
Consider the following example, the methods should return the same result.
![Screen Shot 2020-07-30 at 9 20 55 AM](https://user-images.githubusercontent.com/52430951/88934331-2051d900-d246-11ea-98f3-31f271724fd7.png)

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more and fix the things.